docs/design-ux/sharing.md

```diff
@@ -0,0 +1,104 @@
+# Sharing
+
+The sharing UX needs to balance simplicity with flexibility. Users need to be able to clearly tell who can access which
+documents and easily modify that access. They also need to be able to accomplish a wide variety of sharing-related
+tasks, including:
+
+* sharing individual documents or whole folders
+* sharing with readonly vs read/write access
+* managing who is billed for shared documents
+* linking to shared documents from shared documents and from unshared documents We evaluate 3 design alternatives
+  according to their suitability for individuals, teams, and enterprises. Individuals are users who store thoughts in
+  lockbook and occassionally want to share a file with another user. Teams are small groups of users who generally want
+  someplace to store personal things and someplace to store team things, which are shared with the whole team.
+  Enterprises are large organizations that want to control how things are shared in sophisticated ways, such as having
+  teams with subteams that have at least as much access. The designs we consider are:
+
+1. Root sharing
+2. Folder sharing
+3. Policy Sharing
+
+## 1. Root Sharing
```

Comment:
Honestly vault-sharing would simplify *SO MUCH* of core, maybe it's the complexity wins we need to get off the ground.

Reply:
This also performs well for links, every document has an absolute path within lockbook.

Reply:
How can we address the concerns for the individuals use cases? Can we have vault sharing as described + individual document sharing by permalink that shows up in a 'shared with me' section or some such hack? Weird to have 2 sharing models but they're both simple enough that it's still a simplification and we can implement one at a time to get off the ground

Reply:
Yeah I think for individual use cases, being able to share one-off documents is probably sufficient. We'll need to give some thought to what the receiver experiences and how they keep track of things that are shared with them outside of vaults. 

Google drive just does the "shared with me" sort of region, the thing that helps you organize that is the context around who shared it with you and when. That's probably easy for us to do.

The "add to my drive" sort of experience is probably what we want to give more thought to. I think, like you're suggesting this will probably be our version of symlinks. A symlink is a doc which points to a UUID of a file. It's just a navigation thing, probably. This is perhaps a new `file_type`, how exciting.